samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -134,7 +136,8 @@ public final void init(InternalTaskContext internalTaskContext) {
     this.taskModel = taskContext.getTaskModel();
     this.callbackScheduler = taskContext.getCallbackScheduler();
     handleInit(context);
-    this.elasticityFactor = new JobConfig(context.getJobContext().getConfig()).getElasticityFactor();
+    this.elasticityFactor = new JobConfig(config).getElasticityFactor();
+    this.operatorThreadPool = context.getTaskContext().getOperatorExecutor();
```

Comment:
Should this thread pool also be used in StreamOperatorTask?

Reply:
That is out of scope for this PR as there are other considerations w/ the change to maintain backward semantics and also control which thread pool is used.

Reply:
I do have a TODO in `StreamOperatorTask` to tackle this. 
My initial take on doing that refactor too also resulted in fixing a bunch of tests and hence resorted to tackling it separately and keep the scope of this PR to `OperatorImpl`